### DISCUSSION TITLE
Help needed in setting up TCAT commissioner

### DISCUSSION THREAD

**Discussion by RexPat** [openthread/openthread#9855] [Q&A]
We are attempting to setup TCAT commissioner in RPi, referring [README.md](https://github.com/openthread/openthread/blob/main/tools/tcat_ble_client/README.md) 
`poetry install` command was successful, but the following command ended up in the error, below is the screenshot of CLI

![image](https://github.com/openthread/openthread/assets/124576857/58487e9c-f632-41db-9c40-46bbb6b68d43)

The import is being refered in [base_commands.py#L34](https://github.com/openthread/openthread/blob/49c59ec519cc8b49dd58978d1bc80b7ae7ba88d0/tools/tcat_ble_client/cli/base_commands.py#L34) and some other places too. 

Error in above due to a missing reference. when checked there is no file named tlv.py was found in [tlv](https://github.com/openthread/openthread/tree/main/tools/tcat_ble_client/tlv) folder.

Am i missing any step??

**Comment by karthick-grl** [comment]
@pjasinski990 Would you be able to help in this?

**Comment by maciejbaczmanski** [comment]
CC @canisLupus1313

**Comment by maciejbaczmanski** [comment]
https://github.com/openthread/openthread/pull/9857 should fix the issue

**Comment by RexPat** [comment]
@maciejbaczmanski Thank you. The PR fixed the issue. 

When I tried running the commissioner code on a Raspberry Pi 4 B with Raspbian OS built from [ot-br-posix](https://github.com/openthread/ot-br-posix), it failed due to issues with Bluetooth drivers. Attempts to install the drivers were unsuccessful as well. However, while using the latest official Raspbian OS (without otbr posix environment), the built-in Bluetooth was enabled, and I was able to discover the "Thread BLE" client dongle and other BLE devices too. Does the commissioner code require otbr-agent or ot-br-posix environment to work? What is the recommended environment to make the TCAT commissioner work?

**Comment by canisLupus1313** [comment]
@RexPat the commissioner doesn't require any of OTBR env.  In fact it just requires Bluetooth capabilities on you device. We have been So all in all You can have two approaches in the cert environment. Either put a BLE interface accessible from host running the test env and use the commissioner natively, or put raspberry Pi with working BLE and  commissioner into the test setup and communicate with it via eg. ssh.
